service/resource/legacyv2/adapter/internet_gateway.go

```diff
@@ -0,0 +1,26 @@
+package adapter
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+// template related to this adapter: service/templates/cloudformation/internet_gateway.yaml
+
+type internetGatewayAdapter struct {
+	VPCID string
+}
```

Comment:
The adapter is used to render data in the template. For the inet gateway we just need the VPC ID.